Bottom: e3fcd2e2225e35f6bb72bf1f208e2fa045c8d14a
Top:    9d0506751ef06d66715efc9f95308a2928073752
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2008-10-13 21:00:00 +0100

Try optimising polygon operations by keeping an rtree with their contours

InsertHoles() already creates an rtree of contours to do its internal
processing, and it is possible that intersect() could be sped up by
its existance as well.




---

diff --git a/src/polyarea.h b/src/polyarea.h
index 42acc1a..819b074 100644
--- a/src/polyarea.h
+++ b/src/polyarea.h
@@ -128,6 +128,7 @@ struct POLYAREA
 {
     POLYAREA *f, *b;
     PLINE *contours;
+    void *contour_tree;
 };
 
 BOOLp poly_M_Copy0(POLYAREA ** dst, const POLYAREA * srcfst);
